### ISSUE TITLE
sql/distsqlrun: nodedialer: "connection established" log spam

### ISSUE THREAD

**Issue by bdarnell** [cockroachdb/cockroach#28165]
We see a lot of log spam related to `nodedialer`:

```
I180801 20:32:42.762866 321630255 rpc/nodedialer/nodedialer.go:90  [n1] connection to n6 established
I180801 20:32:42.883486 321630241 rpc/nodedialer/nodedialer.go:90  [n1] connection to n6 established
I180801 20:32:43.233148 321630356 rpc/nodedialer/nodedialer.go:90  [n1] connection to n3 established
I180801 20:32:43.427715 321630427 rpc/nodedialer/nodedialer.go:90  [n1] connection to n5 established
I180801 20:32:43.508489 321630498 rpc/nodedialer/nodedialer.go:90  [n1] connection to n2 established
I180801 20:32:43.859990 321630745 rpc/nodedialer/nodedialer.go:90  [n1] connection to n2 established
I180801 20:32:43.994620 321630861 rpc/nodedialer/nodedialer.go:90  [n1] connection to n4 established
I180801 20:32:44.029338 33 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52134,user=root] connection to n5 established
I180801 20:32:44.072335 66 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52134,user=root] connection to n3 established
I180801 20:32:44.131726 31 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52124,user=root] connection to n3 established
I180801 20:32:44.176682 67 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52134,user=root] connection to n3 established
I180801 20:32:44.233832 68 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52132,user=root] connection to n3 established
I180801 20:32:44.247071 74 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52134,user=root] connection to n3 established
I180801 20:32:44.269991 75 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52124,user=root] connection to n3 established
I180801 20:32:44.334859 71 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52134,user=root] connection to n5 established
I180801 20:32:44.367074 321631097 rpc/nodedialer/nodedialer.go:90  [n1] connection to n3 established
I180801 20:32:44.388427 70 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52134,user=root] connection to n3 established
I180801 20:32:44.504636 69 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52134,user=root] connection to n2 established
I180801 20:32:44.521339 77 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52124,user=root] connection to n2 established
I180801 20:32:44.592831 321631280 rpc/nodedialer/nodedialer.go:90  [n1] connection to n4 established
I180801 20:32:44.600853 72 rpc/nodedialer/nodedialer.go:90  [n1,client=127.0.0.1:52134,user=root] connection to n6 established
```

From a quick look around, this line looks suspicious:

https://github.com/cockroachdb/cockroach/blob/86c981813e7c9160a75936c3946e565dcdab044c/pkg/sql/distsqlrun/server.go#L408

We need to hoist that up and reuse a nodedialer instead of creating a new one for each distsql flow. That should solve the worst of the problem, but it looks like we probably need some more refactoring to change the 3-4 remaining nodedialers to a singleton.

**Comment by bdarnell** [comment]
@m-schneider this is the issue I mentioned this morning.

**Comment by knz** [comment]
@asubiotto @jordanlewis can you also have a look into this / assist masha as needed?

**Comment by knz** [comment]
Also hit by user @isoos on gitter.

**Comment by m-schneider** [comment]
Some of it should be fixed by #28531. There are only a couple of places I didn't pass a node dialer because that would have been way out of scope of that PR.
